aws-android-sdk-pinpoint/src/main/java/com/amazonaws/mobileconnectors/pinpoint/targeting/notification/NotificationClient.java

```diff
@@ -284,16 +284,27 @@ public String getChannelType() {
      * local broadcast is sent.
      *
      * @param notificationDetails the notification message received by the device's messaging service
-     * @return {@link NotificationClient.CampaignPushResult}.
+     * @return {@link PinpointPushResult}.
+     *
+     */
+    public PinpointPushResult handlePushNotification(final NotificationDetails notificationDetails) {
+        return notificationClientBase.handlePushNotification(notificationDetails);
+    }
+
+    /**
+     * @deprecated see {@link #handlePushNotification(NotificationDetails)}
      */
-    public CampaignPushResult handleCampaignPush(final NotificationDetails notificationDetails) {
-        return notificationClientBase.handleCampaignPush(notificationDetails);
+    @Deprecated
+    public PinpointPushResult handleCampaignPush(final NotificationDetails notificationDetails) {
```

Comment:
Since the return type has changed, anyway, the `@Deprecated` isn't helping much. We might as well just delete it.

Reply:
So here's what I did: I added the old enum back and changed `handleCampaignPush` to return it. Still marked it as deprecated.

The new `handlePushNotification` method returns the new enum type with a more generic name. 

I'm going to get an API review in the books to go over this.